### PR TITLE
Feature/glassmorphism

### PR DESCRIPTION
**Pull request by vatsaltanna**

### CODE REVIEW

File: example/lib/main.dart [hunk 1]
```diff
@@ -18,101 +18,190 @@ class MySampleState extends State<MySample> {
   String cardHolderName = '';
   String cvvCode = '';
   bool isCvvFocused = false;
+  bool useGlassMorphism = false;
+  bool useBackgroundImage = false;
   final GlobalKey<FormState> formKey = GlobalKey<FormState>();
 
   @override
   Widget build(BuildContext context) {
+    final OutlineInputBorder border = OutlineInputBorder(
```
Comment (DevarshRanpara) [inline example/lib/main.dart:27]: Can we define `border` in initState?

File: example/lib/main.dart [hunk 2]
```diff
@@ -18,101 +18,190 @@ class MySampleState extends State<MySample> {
...
+          decoration: BoxDecoration(
+            image: !useBackgroundImage
+                ? const DecorationImage(
+                    image: ExactAssetImage('assets/bg.png'),
+                    fit: BoxFit.fill,
+                  )
+                : null,
+            color: Colors.black,
+          ),
+          child: SafeArea(
```
Comment (DevarshRanpara) [inline example/lib/main.dart:51]: Can we add `SafeArea` at top level in `body`?

File: example/lib/main.dart [hunk 3]
```diff
@@ -18,101 +18,190 @@ class MySampleState extends State<MySample> {
...
-                      ElevatedButton(
-                        style: ElevatedButton.styleFrom(
-                          shape: RoundedRectangleBorder(
-                            borderRadius: BorderRadius.circular(8.0),
+                        ElevatedButton(
+                          style: ElevatedButton.styleFrom(
+                            shape: RoundedRectangleBorder(
+                              borderRadius: BorderRadius.circular(8.0),
+                            ),
+                            primary: const Color(0xff1b447b),
```
Comment (DevarshRanpara) [inline example/lib/main.dart:177]: Can we use constant for this color code?

File: lib/credit_card_widget.dart [hunk 4]
```diff
@@ -495,6 +588,84 @@ class _CreditCardWidgetState extends State<CreditCardWidget>
   }
 }
 
+class _CardBackground extends StatelessWidget {
```
Comment (DevarshRanpara) [inline lib/credit_card_widget.dart:591]: Can we create separate file for this class?

File: lib/credit_card_widget.dart [hunk 5]
```diff
@@ -495,6 +588,84 @@ class _CreditCardWidgetState extends State<CreditCardWidget>
...
+                      ? ((screenWidth - 32) * 0.5714)
+                      : screenHeight / 2),
+            ),
+          ),
+      ],
+    );
+  }
+}
+
 class AnimationCard extends StatelessWidget {
```
Comment (DevarshRanpara) [inline lib/credit_card_widget.dart:669]: Can we create separate file for this class?

File: lib/credit_card_widget.dart [hunk 6]
```diff
@@ -643,6 +814,78 @@ class MaskedTextController extends TextEditingController {
   }
 }
 
+class _GlassmorphicBorder extends StatelessWidget {
```
Comment (DevarshRanpara) [inline lib/credit_card_widget.dart:817]: Can we create separate file for this class?

**Review by DevarshRanpara — CHANGES_REQUESTED** [review]

File: example/lib/main.dart (hunk 2 above)
Comment (vatsaltanna) [reply example/lib/main.dart:51]: we are using background image in top of this so if safe area will be the parent of it then UI will be messed up.

Comment (DevarshRanpara) [reply example/lib/main.dart:51]: Okay got it.

File: lib/credit_card_background.dart [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+                        backgroundImage!,
+                      ),
+                      fit: BoxFit.fill,
+                    )
+                  : null,
+            ),
+            width: width ?? screenWidth,
+            height: height ??
+                (orientation == Orientation.portrait
+                    ? (((width ?? screenWidth) - 32) * 0.5714)
```
Comment (DevarshRanpara) [inline lib/credit_card_background.dart:56]: What is 32 and 0.5714?

Comment (vatsaltanna) [reply lib/credit_card_background.dart]: Its credit card padding and aspect ratio respectively. I have moved them into the constant files.

**Review by DevarshRanpara — APPROVED** [review]
LGTM.

**Pull request merged**